mlflow/pyfunc/model.py

```diff
@@ -0,0 +1,333 @@
+                      ``mlflow.pyfunc.model.DEFAULT_CONDA_ENV`` environment will be added to the
+                      model.
+    :param code_paths: A list of paths to Python file dependencies that are required by
+                       instances of ``model_class``.
+    :param mlflow_model: The model configuration to which to add the ``mlflow.pyfunc`` flavor.
+    """
+    if os.path.exists(path):
+        raise MlflowException(
+                message="Path '{}' already exists".format(path),
+                error_code=INVALID_PARAMETER_VALUE)
+    os.makedirs(path)
+
+    custom_model_config_kwargs = {}
+
+    if artifacts is not None and len(artifacts) > 0:
+        saved_artifacts_config = {}
+        with TempDir() as tmp_artifacts_dir:
+            tmp_artifacts_config = {}
+            saved_artifacts_dir_subpath = "artifacts"
+            for artifact_name, artifact_uri in artifacts.items():
+                tmp_artifact_path = _download_artifact_from_uri(
+                    artifact_uri=artifact_uri, output_path=tmp_artifacts_dir.path())
+                tmp_artifacts_config[artifact_name] = tmp_artifact_path
+                saved_artifact_subpath = os.path.join(
+                    saved_artifacts_dir_subpath,
+                    os.path.relpath(path=tmp_artifact_path, start=tmp_artifacts_dir.path()))
+                saved_artifacts_config[artifact_name] = {
+                    CONFIG_KEY_ARTIFACT_RELATIVE_PATH: saved_artifact_subpath,
+                    CONFIG_KEY_ARTIFACT_URI: artifact_uri,
+                }
+
+            _validate_artifacts(tmp_artifacts_config)
+            shutil.move(tmp_artifacts_dir.path(), os.path.join(path, saved_artifacts_dir_subpath))
+        custom_model_config_kwargs[CONFIG_KEY_ARTIFACTS] = saved_artifacts_config
+
+    if parameters is not None and len(parameters) > 0:
+        saved_parameters_dir_subpath = "parameters"
+        os.makedirs(os.path.join(path, saved_parameters_dir_subpath))
+        saved_parameters_config = {}
+        for parameter_name, parameter_py_obj in parameters.items():
+            saved_parameter_subpath = os.path.join(
+                saved_parameters_dir_subpath, "{param_name}.pkl".format(param_name=parameter_name))
+            with open(os.path.join(path, saved_parameter_subpath), "wb") as out:
+                cloudpickle.dump(parameter_py_obj, out)
+            saved_parameters_config[parameter_name] = saved_parameter_subpath
+        custom_model_config_kwargs[CONFIG_KEY_PARAMETERS] = saved_parameters_config
+
+    saved_model_class_config = {}
+    if inspect.isclass(model_class):
+        saved_model_class_subpath = "model_class.pkl"
+        with open(os.path.join(path, saved_model_class_subpath), "wb") as out:
+            cloudpickle.dump(model_class, out)
+        saved_model_class_config[CONFIG_KEY_MODEL_CLASS_PATH] = saved_model_class_subpath
+    else:
+        saved_model_class_config[CONFIG_KEY_MODEL_CLASS_NAME] = model_class
```

Comment:
this sounds a little confusing: `inspect.isclass(model_class) == False` here so not clear why `model_class` is saved as `CONFIG_KEY_MODEL_CLASS_NAME`

Reply:
Agreed! I've added comments during the model class serialization and deserialization procedures to explain the logical flow. In this case, `inspect.isclass(model_class) == False` implies that `model_class` is not a Python class object. Therefore, we assume that it's the fully-qualified name of a Python class and save it using the key `CONFIG_KEY_MODEL_CLASS_NAME`.

Reply:
Update: I've removed the string-based definition of `model_class`. Now, it must always be a Python class object. This removes the confusion you identified.